charmtools/build/builder.py

```diff
@@ -904,11 +905,17 @@ def main(args=None):
                         "from the interface service.")
     parser.add_argument('-n', '--name',
                         help="Build a charm of 'name' from 'charm'")
-    parser.add_argument('-r', '--report', action="store_true",
+    parser.add_argument('-r', '--report', action="store_true", default=True,
                         help="Show post-build report of changes")
+    parser.add_argument('-R', '--no-report', action="store_false",
```

Comment:
-r and -R seem to be contradictory; how about we kill the -R codepath and just use -r=True|False instead?

Reply:
I tried to figure out a way to implement the `-r[=true/|false]` pattern but the case that always breaks is the current behavior that many may be using:

```
$ charm build -r ./src
```

There's no way with argparse to say "consume an arg, but only if it matches one of these choices".  It will either not consume any args, always consume an arg, or try to consume an arg but blow up if it doesn't match one of the choices.

Additionally, I found at least one example with [`git -p` and `git -P`](https://git-scm.com/docs/git#Documentation/git.txt--p) which have the same contradictory meaning, and I know I've seen other cases and not found it terribly confusing.  In general, and in the case of this change, the behavior is what I would expect: either of the options can be given, even multiple times, and the last one present on the command line takes precedence.

Reply:
@johnsca i appreciate the attempt.  I didn't intend for you to bend over backwards, but it sure would have been nice if "honor a bool value" would *just work* for argparse optargs.  Anywho, I'm annoyed but not blocking contradictory flags; as you said, last in gets the cake, and i think that's a reasonable UX that charm builders can shoulder.

All that said, given that `--report` is now the default, I would ask you to consider #570.  That is, deprecate `-r` with some stderr message that says `-r is default and therefore deprecated; use -R to prevent charm build reports`.